### PR TITLE
Reset handler cleanup fixes for audio processing components

### PR DESCRIPTION
**Pull request by singalsu**
The components FIR, IIR, SRC, TDFB, and volume have some state variables those were not treated as required in reset operation description in component.h. This pull request removes some unnecessary state variables from component by converting them to local or adds reset code.

This contributes to issue #4488 .

Edit: I've removed volume component from this PR. The volume state can't be reset in reset() comp op without causing volume levels test to fail.

### CODE REVIEW

**Review by cujomalainey — APPROVED** [review]
Thanks for fixing this up :)

File: src/audio/volume/volume.c [hunk 1]
```diff
@@ -292,6 +292,27 @@ static void volume_ramp(struct comp_dev *dev)
...
+static void reset_state(struct vol_data *cd)
+{
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
+				      cd->vol_min);
+		cd->tvolume[i] = cd->volume[i];
+		cd->mvolume[i] = cd->volume[i];
+		cd->muted[i] = false;
```
Comment (cujomalainey) [inline src/audio/volume/volume.c:304]: oh one thing, could this possibly unmute a channel through a close/open procedure even in control is still muted?

File: src/audio/volume/volume.c [hunk 2]
```diff
@@ -292,6 +292,27 @@ static void volume_ramp(struct comp_dev *dev)
 	vol_sync_host(dev, cd->channels);
 }
 
+static void reset_state(struct vol_data *cd)
+{
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
+				      cd->vol_min);
```
Comment (lyakh) [inline src/audio/volume/volume.c:301]: I presume `VOL_ZERO_DB` cannot be outside of the `min-max` range?

Comment (singalsu) [reply src/audio/volume/volume.c:301]: Nothing prevents topology to define e.g. volume scale from +10 dB to +50 dB. In that case volume would default to +10 dB until kernel sends a volume cmd(). Or if for some device safe (speaker with power limit) volume scale is -60 dB to -20 dB then volume would default to -20 dB that is within allowed range.

File: src/audio/volume/volume.c (hunk 1 above)
Comment (singalsu) [reply src/audio/volume/volume.c:304]: That was my worry too. Seem like CI does not run yet sof-test/test-case/check-volume-levels.sh. I just ran the  testmanually and this PR failed while git main branch passes. The kernel driver apparently does not re-apply all the controls if reset state clearing impacts. I think I should remove the volume commit from this PR and study if it's a bug.

**Review by lgirdwood — APPROVED** [review]

File: src/audio/volume/volume.c (hunk 1 above)
Comment (cujomalainey) [reply src/audio/volume/volume.c]: yes i agree. I think control data is outside the scope of reset. I think its pipeline state that is only supposed to be reset

**Comment by cujomalainey** [comment]
did you accidentally rebase the volume change out of the branch?

File: src/audio/volume/volume.c (hunk 2 above)
Comment (lyakh) [reply src/audio/volume/volume.c]: ah, sure, you're right. So effectively this selects the closest to 0dB allowed value.

**Comment by singalsu** [comment]
> did you accidentally rebase the volume change out of the branch?

No, I did it intentionally due to risk of breaking things while I will be away on vacation for 2 weeks. I'm testing more and will make a separate PR for it.

**Pull request merged**